core/regions/src/main/java/software/amazon/awssdk/regions/PartitionMetadata.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.regions;
+
+/**
+ * Metadata about a partition such as aws or aws-cn.
+ *
+ * <p>This is useful for building meta-functionality around AWS services. Partition metadata helps to provide
+ * data about regions which may not yet be in the endpoints.json file but have a specific prefix.</p>
+ */
+public interface PartitionMetadata {
```

Comment:
checktyle might fail because of missing sdk annotation

Reply:
Wasn't failing for me locally but might have an old version of checkstyle. Will add and run checkstyle once I merge from master.